scrape/scrape_test.go

```diff
@@ -3168,6 +3157,52 @@ func TestScrapeReportLimit(t *testing.T) {
 	require.True(t, found)
 }
 
+func TestScrapeUTF8(t *testing.T) {
+	s := teststorage.New(t)
+	defer s.Close()
+	model.NameValidationScheme = model.UTF8Validation
```

Comment:
Isn't the idea here to test that we do UTF-8 validation _despite_ not setting it globally in the `model` package?
In other words: Shouldn't we leave `model.NameValidationScheme` alone in this test?

Reply:
Both configs must be set for UTF8 scraping+writing to work ([docs](https://github.com/prometheus/prometheus/blob/main/docs/feature_flags.md#utf-8-name-support)):
1. `--enable-feature=utf8-names` flag, which sets `model.NameValidationScheme = model.UTF8Validation` [here](https://github.com/prometheus/prometheus/blob/569b6abfa3f876419879989d9e17d9a4f3cfc01b/cmd/prometheus/main.go#L255-L257) .
2. `metric_name_validation_scheme: utf8` in the scrape configs. This was the one with the bug

I don't fully understand why the configuration is split this way, but I'm sure there's a good reason.

Reply:
Side note: I can confirm this test fails if I revert the change

Reply:
Ah, perfect then. I must have looked at a wrong state of my git repo…